SchoolNetAssociation.IntegrationTest/Controllers/SchoolDistrictControllerIntegrationTests.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System.Net;
+using System.Net.Http.Json;
+using Bogus;
+using FluentAssertions;
+using Microsoft.AspNetCore.Mvc.Testing;
+using SchoolNetAssociation.Domain.Entities;
+
+namespace SchoolNetAssociation.IntegrationTest
+{
+    public class SchoolDistrictControllerTests : IClassFixture<WebApplicationFactory<Program>>
+    {
+        private readonly HttpClient _httpClient;
+        private readonly Faker<SchoolDistrict> _schoolDistrictFaker;
+        private readonly Faker _faker;
+
+        const string ClientBaseUrl = "api/SchoolDistrict";
+
+        public SchoolDistrictControllerTests(WebApplicationFactory<Program> factory)
+        {
+            _httpClient = factory.CreateClient();
+            _faker=new Faker();
+            _schoolDistrictFaker = new Faker<SchoolDistrict>()
+                .RuleFor(sd => sd.Name, f => f.Company.CompanyName())
+                .RuleFor(sd => sd.Description, f => f.Lorem.Sentence())
+                .RuleFor(sd => sd.City, f => f.Address.City())
+                .RuleFor(sd => sd.Superintendent, f => f.Name.FullName())
+                .RuleFor(sd => sd.IsPublic, f => f.Random.Bool())
+                .RuleFor(sd => sd.NumberOfSchools, f => f.Random.Int(1, 1000));
+        }
+
+        [Fact]
+        public async Task GetAllSchoolDistrictsAsync_ShouldReturnOkWithSchoolDistricts()
+        {
+            // Act
+            var response = await _httpClient.GetAsync(ClientBaseUrl);
+            var result = await response.Content.ReadFromJsonAsync<SchoolDistrict[]>();
+
+            // Assert
+            response.StatusCode.Should().Be(HttpStatusCode.OK);
+            result.Should().NotBeEmpty();
```

Comment:
I think I have mentioned this multiple times before testing for not empty is not good enough